src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -76,22 +80,26 @@ public ComplaintResource(ComplaintRepository complaintRepository, ResultReposito
             throw new BadRequestAlertException("A complaint can be only associated to a result", ENTITY_NAME, "noresultid");
         }
 
-        Long resultId = complaint.getResult().getId();
-        String submissorName = principal.getName();
-        User originalSubmissor = userRepository.findUserByResultId(resultId);
-
-        if (!originalSubmissor.getLogin().equals(submissorName)) {
-            throw new BadRequestAlertException("You can create a complaint only about a result you submitted", ENTITY_NAME, "differentuser");
+        if (complaintRepository.findByResult_Id(complaint.getResult().getId()).isPresent()) {
+            throw new BadRequestAlertException("A complaint for this result already exists", ENTITY_NAME, "complaintexists");
         }
 
         // Do not trust user input
-        Optional<Result> originalResultOptional = resultRepository.findById(resultId);
+        Result originalResult = resultRepository.findById(complaint.getResult().getId())
+                .orElseThrow(() -> new BadRequestAlertException("The result you are referring to does not exist", ENTITY_NAME, "resultnotfound"));
+        User originalSubmissor = originalResult.getParticipation().getStudent();
+        Long courseId = originalResult.getParticipation().getExercise().getCourse().getId();
+
+        long numberOfUnacceptedComplaints = complaintRepository.countUnacceptedComplaintsByStudentIdAndCourseId(originalSubmissor.getId(), courseId);
+        if (numberOfUnacceptedComplaints >= MAX_COMPLAINT_NUMBER_PER_STUDENT) {
+            throw new BadRequestAlertException("You cannot have more than " + MAX_COMPLAINT_NUMBER_PER_STUDENT + " open or rejected complaints at the same time.", ENTITY_NAME,
+                    "toomanycomplaints");
+        }
```

Comment:
I think you are missing the check for how old is the result, since you cannot create complaints for results older than a week

Reply:
Good catch, I added it.